src/interfaces/IBPool.sol

```diff
@@ -54,53 +131,141 @@ interface IBPool is IERC20 {
     uint256 maxPrice
   ) external returns (uint256 tokenAmountIn, uint256 spotPriceAfter);
 
+  /**
+   * @notice Joins a pool providing only a single token, and a specific amount of tokenIn
```

Comment:
Joins a pool providing a single token in, specifying the exact amount of token given